### ISSUE TITLE
Add support for new Azure Database for MySQL - Flexible Server resource

### ISSUE THREAD

**Issue by supertrampr** [gruntwork-io/terratest#1230]
Due to upcoming changes on classic Azure Database for MySQL deployment ([see this for reference](https://learn.microsoft.com/en-us/azure/mysql/single-server/whats-happening-to-mysql-single-server)), I would like to contribute on terratest examples and code for interacting and testing new available resources [**azurerm_mysql_flexible_server**](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server) and [**azurerm_mysql_flexible_database**](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_database). 

A complication that I noticed is that the [azure-sdk-for-go](https://github.com/gruntwork-io/terratest/blob/v0.41.9/go.mod) that Terratest use is very old and does not expose the needed method for the above mentioned resources until version [57.1.0](https://github.com/Azure/azure-sdk-for-go/tree/v57.1.0/profiles/latest/mysql/mgmt/mysqlflexibleservers) so it would be a huge jump for az-sdk-for-go module and packages referencing it. Let me know if I have to file a related issue to this to ask for an upgrade on Terratest go.mod file!

I was not able to find anything related to this yet so I would like to contribute together on this :) Just wanted to ask if there is any constraint on what [AzureRM provider](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs) version or if we are free to use the latest version. 

Also I am new to contributing at open source project so let me know if I missed something out on this issue.

**Comment by denis256** [comment]
Hello,
changes can be separated into multiple steps:
* azure-sdk-for-go dependencies upgrade
* implementation of new Azure Database for MySQL

Will be great if backward compatibility will be kept between upgrades

**Comment by supertrampr** [comment]
> Hello, changes can be separated into multiple steps:
> 
> * azure-sdk-for-go dependencies upgrade
> * implementation of new Azure Database for MySQL
> 
> Will be great if backward compatibility will be kept between upgrades

Yes, I'll check but as far as I know Azure MySQL Single Server will be dismissed on 2024 and the sdk is being updated in parallel with both resource... If is it what you meant and this answer your concern :) Anyway I'll test on my side if upgrading to at least the minimum version required to have flexible servers would not break any other azure component. Thanks for your reply I'll get back to you as soon as I have some updates! Thanks @denis256  I am very glad to be of some kind of help on

**Comment by supertrampr** [comment]
Hey @denis256, just opened pull request https://github.com/gruntwork-io/terratest/pull/1245. Let me know if there is something I need to change from my part!

**Comment by james00012** [comment]
Valid request, but adding new Azure resource modules is not on the maintainer roadmap as Terratest moves to a maintenance posture. We welcome a community PR that adds azurerm_mysql_flexible_server and azurerm_mysql_flexible_database support following the existing Azure module patterns. Closing as not planned.
